### PR TITLE
Fixed IInvokedMethodListener documentation

### LINKED ISSUE #1870: Ambiguous behavior of IInvokedMethodListener
### TestNG Version
6.14.3

IInvokedMethodListener Javadoc states:

> A listener that gets invoked before and after a method is invoked by TestNG.
> This listener will only be invoked for configuration and test methods.

However, it seems like actual behavior is a bit ambiguous (see #1602 and #1317)

### Expected behavior
Listener methods should only be called when a method is **invoked**

### Actual behavior
Even if method is skipped completely (because of a failed configuration for example), listener methods are still called

### Test case sample
Run the following test class with the provided listener
```java
public class SkippedTestMethod {
    @BeforeClass
    public void beforeClass() {
        throw new SkipException("Skippity Skip");
    }

    @Test
    public void testMethod() {
        // shouldn't ever be called
    }
}

public class SkippedTestMethodListener implements IInvokedMethodListener {

    @Override
    public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
        if(method.isTestMethod()) {
            throw new RuntimeException("beforeInvocation should not be called if no actual invocation occurs!");
        }
    }

    @Override
    public void afterInvocation(IInvokedMethod method, ITestResult testResult) {
        if(method.isTestMethod()) {
            throw new RuntimeException("afterInvocation should not be called if no actual invocation occurs!");
        }
    }
}
```

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #1870

Fixes #1870 .

### Did you remember to?

- [ ] Add test case(s)
- [ ] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/main/java/org/testng/IInvokedMethodListener.java [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 /**
  * A listener that gets invoked before and after a method is invoked by TestNG. This listener will
- * only be invoked for configuration and test methods.
+ * be invoked for configuration and test methods irrespective of whether they passe/fail or
+ * get skipped.
```
Comment (juherr) [inline src/main/java/org/testng/IInvokedMethodListener.java]: Not sure for skipped. I will let @cbeust takes the decision.

Comment (krmahadevan) [reply src/main/java/org/testng/IInvokedMethodListener.java]: @juherr - The documentation is reflecting the current behavior

Comment (juherr) [reply src/main/java/org/testng/IInvokedMethodListener.java]: I think this PR should be merged with #1884

**Review by juherr — APPROVED** [review]

**Comment by krmahadevan** [comment]
Based on an email from @cbeust - merging this PR since @juherr has helped approved it.

File: src/main/java/org/testng/IInvokedMethodListener.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/IInvokedMethodListener.java]: I will come back and alter the javadocs accordingly once I take up https://github.com/cbeust/testng/issues/1892

**Pull request merged**